x/ccv/provider/keeper/proposal.go

```diff
@@ -311,6 +311,37 @@ func (k Keeper) ConsumerAdditionPropsToExecute(ctx sdk.Context) []types.Consumer
 	return propsToExecute
 }
 
+// GetAllConsumerAdditionProps returns all consumer addition proposals separated into matured and pending.
+func (k Keeper) GetAllConsumerAdditionProps(ctx sdk.Context) types.ConsumerAdditionProposals {
+	props := types.ConsumerAdditionProposals{}
+
+	store := ctx.KVStore(k.storeKey)
+	iterator := sdk.KVStorePrefixIterator(store, []byte{types.PendingCAPBytePrefix})
+	defer iterator.Close()
+
+	if !iterator.Valid() {
+		return props
+	}
```

Comment:
Is this necessary? If it's not Valid won't it skip the loop below anyway?

Reply:
Yea, can probably remove this

Reply:
`Done.`